src/plugins/management/public/management_service.ts

```diff
@@ -48,8 +48,9 @@ export class ManagementService {
       return newSection;
     };
   }
-  private getSection(sectionId: ManagementSection['id']) {
-    return this.sections.find(section => section.id === sectionId);
+
+  private getSection(sectionId: ManagementSectionId): ManagementSection {
```

Comment:
Does anybody have any suggestions on what I could do to convince TypeScript that this function will always return a `ManagementSection`, and not `undefined`? TS check shows this error everywhere this method is invoked:

```
 Object is possibly 'undefined'.
```

Reply:
We could consider throwing if `find` returns `undefined`. That would give us both transpile-time and runtime guarantees

Reply:
See my previous comment. I consider this a flaw in the API. I think `throw`ing will work in the mean time.

Reply:
Thanks Matt! I updated the code to throw.